### PR TITLE
CarbonRange

### PR DESCRIPTION
**Pull request by 20TRIES**
Hey, i needed a way of describing a date range in an application i was writing and thought it may be a useful extension here.

Its not finished, so there still needs to be some more testing and the code could be a little clearer/clearer; however i wanted to see if you thought it may be a useful addition before going all out with the idea.

It would allow stuff like the following:

`$week_start = Carbon::now()->startOfWeek();`

`$week_end = Carbon::now()->endOfWeek();`

`$date_range = CarbonRange::between($week_start, $week_end);`

`$date_range->forHuman();`

With a result of "This Week"

### CODE REVIEW

**Comment by dereuromark** [comment]
What is the difference of Range vs the existing Interval?

**Comment by 20TRIES** [comment]
I was thinking that while the interval is say, 3 months. The range would have a "context", so it would have a start and end date.

I haven't used the interval class too much in the past; maybe if the implementation is very similar this would be better as an extension of the interval class instead?

File: src/Carbon/CarbonRange.php [hunk 1]
```diff
@@ -0,0 +1,330 @@
+<?php namespace Carbon;
+
+/**
+ * An object for interacting with a date range.
+ *
+ * @package Carbon
```
Comment (GrahamCampbell) [inline src/Carbon/CarbonRange.php]: this should be ommited

File: src/Carbon/CarbonRange.php [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+ * @author Marcus T <marcust261@icloud.com>
```
Comment (GrahamCampbell) [inline src/Carbon/CarbonRange.php]: this should be ommited

File: src/Carbon/CarbonRange.php [hunk 3]
```diff
@@ -0,0 +1,330 @@
+<?php namespace Carbon;
```
Comment (GrahamCampbell) [inline src/Carbon/CarbonRange.php]: incorrect cs

Comment (GrahamCampbell) [inline src/Carbon/CarbonRange.php]: should be:

```php
<?php
/*
 * This file is part of the Carbon package.
 *
 * (c) Brian Nesbitt <brian@nesbot.com>
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 */
namespace Carbon;

```

**Pull request closed without merging**